### PR TITLE
feat: add querying support for deposits and deleted DIDs

### PR DESCRIPTION
**Pull request by Diiaablo95**
# Add querying support for deposits and deleted DIDs

This PR adds:
- support to query the amount of KILTs needed to deposit before creating an attestation, a delegation, or a DID
- support to query all the deleted DID identifiers and also to check whether a given DID identifier has been deleted (or blacklisted).

For both features, integration tests have been added.

## Linked components

- Node commit with which this PR has been tested to work: https://github.com/KILTprotocol/mashnet-node/commit/2e937de64c2c0979d2f1b74b8faa2a038707828c
- Fixes https://github.com/KILTprotocol/ticket/issues/1657
- Fixes https://github.com/KILTprotocol/ticket/issues/1639

Blocking https://github.com/KILTprotocol/sdk-js/pull/431.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/core/src/attestation/Attestation.chain.ts [hunk 1]
```diff
@@ -155,3 +156,13 @@ export async function reclaimDeposit(
   )
   return tx
 }
+
+async function queryRawDepositAmount(): Promise<U128> {
```
Comment (Dudleyneedham) [inline packages/core/src/attestation/Attestation.chain.ts:160]: Is there a use-case to `queryRawDepositAmount`, it seems most will use the `queryDepositAmount`?

File: packages/core/src/delegation/DelegationNode.chain.ts [hunk 2]
```diff
@@ -179,3 +180,13 @@ export async function getAttestationHashes(
   >(id)
   return decodeDelegatedAttestations(encodedHashes)
 }
+
+async function queryRawDepositAmount(): Promise<U128> {
```
Comment (Dudleyneedham) [inline packages/core/src/delegation/DelegationNode.chain.ts:184]: Again the same as above.

File: packages/did/src/Did.chain.ts [hunk 3]
```diff
@@ -55,6 +55,17 @@ export async function queryDidEncoded(
   return api.query.did.did<Option<IDidChainRecordCodec>>(didIdentifier)
 }
 
+// Query ALL deleted DIDs, which can get quite time consuming.
```
Comment (Dudleyneedham) [inline packages/did/src/Did.chain.ts:58]: `// Query ALL deleted DIDs this is time-consuming.`

File: packages/did/src/Did.chain.ts [hunk 4]
```diff
@@ -88,6 +99,11 @@ export async function queryEndpointsCountsEncoded(
   return api.query.did.didEndpointsCount<u32>(didIdentifier)
 }
 
+async function queryDepositAmountEncoded(): Promise<U128> {
```
Comment (Dudleyneedham) [inline packages/did/src/Did.chain.ts:102]: Same as above. If it is needed, then this should be renamed to follow the convention of being `raw` amount.

**Review by Dudleyneedham — COMMENTED** [review]
Minor suggestions to change naming

File: packages/core/src/attestation/Attestation.chain.ts (hunk 1 above)
Comment (Diiaablo95) [reply packages/core/src/attestation/Attestation.chain.ts:160]: Not really, but I did it to maintain consistency with the other functions as well. The lowest level fetching directly from the blockchain, and a higher level decoding those values as needed.

File: packages/did/src/Did.chain.ts (hunk 4 above)
Comment (Diiaablo95) [reply packages/did/src/Did.chain.ts:102]: You are right. I will rename everything as encoded.

File: packages/core/src/attestation/Attestation.chain.ts (hunk 1 above)
Comment (Diiaablo95) [reply packages/core/src/attestation/Attestation.chain.ts:160]: Indeed, the function is not even exported, so not usable outside of this module.

File: packages/did/src/Did.chain.ts (hunk 3 above)
Comment (Dudleyneedham) [reply packages/did/src/Did.chain.ts]: What you added now reads better.

File: packages/chain-helpers/src/errorhandling/ExtrinsicError.ts [hunk 5]
```diff
@@ -143,7 +143,6 @@ export const ExtrinsicErrors = {
 
 /**
  * @internal
- * PalletIndex reflects the numerical index of a pallet assigned in the chain's metadata.
```
Comment (Dudleyneedham) [inline packages/chain-helpers/src/errorhandling/ExtrinsicError.ts:146]: Please, add back. 👍

File: packages/did/src/Did.chain.ts [hunk 6]
```diff
@@ -55,6 +55,31 @@ export async function queryDidEncoded(
...
+  // Query all the storage keys, and then only take the relevant property, i.e., the encoded DID identifier.
+  return api.query.did.didBlacklist
+    .keys<GenericAccountId[]>()
+    .then((entries) =>
+      entries.map(({ args: [encodedDidIdentifier] }) => encodedDidIdentifier)
+    )
+}
+
+// Returns the raw representation of the storage entry for the given DID identifier.
+export async function queryDidDeletionStatusEncoded(
```
Comment (Dudleyneedham) [inline packages/did/src/Did.chain.ts:70]: Does this need to be exported?

**Review by Dudleyneedham — CHANGES_REQUESTED** [review]
Looking good, just a minor question.

File: packages/chain-helpers/src/errorhandling/ExtrinsicError.ts (hunk 5 above)
Comment (Diiaablo95) [reply packages/chain-helpers/src/errorhandling/ExtrinsicError.ts]: https://github.com/KILTprotocol/sdk-js/pull/430/commits/f531e493e386bb09102141e7c3346b586885ae0c

File: packages/did/src/Did.chain.ts (hunk 6 above)
Comment (Diiaablo95) [reply packages/did/src/Did.chain.ts]: https://github.com/KILTprotocol/sdk-js/pull/430/commits/b55b82314ca92e4bc2c590ddc2b0bad5f3c3dfe1

**Review by tjwelde — APPROVED** [review]
LGTM

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Pull request merged**